src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/NrbfDecoder.cs

```diff
@@ -23,28 +24,40 @@ public static class NrbfDecoder
     /// </summary>
     /// <param name="bytes">The buffer to inspect.</param>
     /// <returns><see langword="true" /> if it starts with NRBF payload header; otherwise, <see langword="false" />.</returns>
-    public static bool StartsWithPayloadHeader(byte[] bytes)
+    public static bool StartsWithPayloadHeader(ReadOnlySpan<byte> bytes)
     {
-#if NET
-        ArgumentNullException.ThrowIfNull(bytes);
-#else
-        if (bytes is null)
+        if (bytes.Length < SerializedStreamHeaderRecord.Size || bytes[0] != (byte)SerializationRecordType.SerializedStreamHeader)
         {
-            throw new ArgumentNullException(nameof(bytes));
+            return false;
         }
-#endif
 
-        return bytes.Length >= SerializedStreamHeaderRecord.Size
-            && bytes[0] == (byte)SerializationRecordType.SerializedStreamHeader
-#if NET
-            && BinaryPrimitives.ReadInt32LittleEndian(bytes.AsSpan(9)) == SerializedStreamHeaderRecord.MajorVersion
-            && BinaryPrimitives.ReadInt32LittleEndian(bytes.AsSpan(13)) == SerializedStreamHeaderRecord.MinorVersion;
-#else
-            && BitConverter.ToInt32(bytes, 9) == SerializedStreamHeaderRecord.MajorVersion
-            && BitConverter.ToInt32(bytes, 13) == SerializedStreamHeaderRecord.MinorVersion;
-#endif
-    }
+        // The header consists of:
+        // - a byte that describes the record type (SerializationRecordType.SerializedStreamHeader)
+        // - four 32 bit integers:
+        //   - root Id (every value is valid)
+        //   - header Id (value is ignored)
+        //   - major version, it has to be equal 1.
+        //   - minor version, it has to be equal 0.
+        // For little endian: 0 * * * * * * * * 1 0 0 0 0 0 0 0
```

Comment:
The wire format is always little endian, correct? So this comment and the code that follows does not make sense to me. If you are reading one byte at a time, there should not be any `if (BitConverter.IsLittleEndian)` conditions.